crates/viewer/re_renderer/shader/world_grid.wgsl

```diff
@@ -0,0 +1,154 @@
+#import <./global_bindings.wgsl>
+
+struct WorldGridUniformBuffer {
+    color: vec4f,
+
+    /// A value of [`super::GridPlane`]
+    orientation: u32,
+
+    /// How far apart the closest sets of lines are.
+    spacing: f32,
+
+    /// How thick the lines are in UI units.
+    thickness_ui: f32,
+}
+
+@group(1) @binding(0)
+var<uniform> config: WorldGridUniformBuffer;
+
+// See world_grid::GridPlane
+const ORIENTATION_XZ: u32 = 0;
+const ORIENTATION_YZ: u32 = 1;
+const ORIENTATION_XY: u32 = 2;
+
+struct VertexOutput {
+    @builtin(position)
+    position: vec4f,
+
+    @location(0)
+    scaled_world_plane_position: vec2f,
+};
+
+const PLANE_SCALE: f32 = 10000.0;
+
+// Spans a large quad where centered around the camera.
+//
+// This gives us the "canvas" to drawn the grid on.
+// Compared to a fullscreen pass, we get the z value (and thus early z testing) for free,
+// as well as never covering the screen above the horizon.
+@vertex
+fn main_vs(@builtin(vertex_index) v_idx: u32) -> VertexOutput {
+    var out: VertexOutput;
+
+    var plane_position = (vec2f(f32(v_idx / 2u), f32(v_idx % 2u)) * 2.0 - 1.0) * PLANE_SCALE;
```

Comment:
Seems like `PLANE_SCALE` is the maximum radius of the area covered by grid lines, i.e. half the distance between the furthest parallel lines?

Do you have a plan for how to handle big scenes (e.g. flight traffic data in meter units)?

Reply:
I'll elaborate in the comment for the plane scale (you're right to point that one out as unclear!): this is the main drawback of the plane approach over the screen space filling one: I have to make up some geometry which then gets a limited size. Putting a too high number here makes things break down because of floating point inaccuracies. But arguably at that point we're potentially doomed either way (see respective ticket on rendering breaking down at some scales)